src/urllib3/poolmanager.py

```diff
@@ -243,22 +282,26 @@ def connection_from_host(self, host, port=None, scheme="http", pool_kwargs=None)
 
         return self.connection_from_context(request_context)
 
-    def connection_from_context(self, request_context):
+    def connection_from_context(
+        self, request_context: Dict[str, Any]
+    ) -> Union[HTTPConnectionPool, HTTPSConnectionPool]:
         """
         Get a :class:`urllib3.connectionpool.ConnectionPool` based on the request context.
 
         ``request_context`` must at least contain the ``scheme`` key and its
         value must be a key in ``key_fn_by_scheme`` instance variable.
         """
-        scheme = request_context["scheme"].lower()
+        scheme = str(request_context["scheme"]).lower()
         pool_key_constructor = self.key_fn_by_scheme.get(scheme)
         if not pool_key_constructor:
             raise URLSchemeUnknown(scheme)
         pool_key = pool_key_constructor(request_context)
 
         return self.connection_from_pool_key(pool_key, request_context=request_context)
 
-    def connection_from_pool_key(self, pool_key, request_context=None):
+    def connection_from_pool_key(
```

Comment:
The `request_context` parameter here can't be `Optional` as it's used immediately below. This is a strange one, this API is only used in one place and in that place `request_context` is required. Maybe it should be required here as well? If we do this we need to mark it in the "breaking change" list as it technically is but almost no one should be using these APIs unless they know what they're doing so it should be fine :)